In-house patch to adjust nova/virt/opts.py to omit the Hyper-V
hypervisor driver as it has unresolved dependencies and to add the
Solaris Zones hypervisor driver, as well as some missing common
components.

--- nova-13.1.0/etc/nova/nova-config-generator.conf.~1~	2016-06-14 08:45:49.000000000 -0700
+++ nova-13.1.0/etc/nova/nova-config-generator.conf	2016-07-25 23:42:03.124618180 -0700
@@ -21,3 +21,10 @@ namespace = oslo.db
 namespace = oslo.middleware
 namespace = oslo.concurrency
 namespace = keystonemiddleware.auth_token
+namespace = castellan.config
+namespace = oslo.db.concurrency
+namespace = oslo.middleware.cors
+namespace = oslo.middleware.sizelimit
+namespace = oslo.middleware.ssl
+namespace = oslo.reports
+namespace = oslo.versionedobjects
--- nova-13.1.0/nova/virt/opts.py.~1~	2016-06-14 08:45:49.000000000 -0700
+++ nova-13.1.0/nova/virt/opts.py	2016-07-25 23:40:46.653786235 -0700
@@ -15,11 +15,6 @@ import itertools
 import nova.conf
 import nova.virt.configdrive
 import nova.virt.disk.vfs.guestfs
-import nova.virt.hyperv.eventhandler
-import nova.virt.hyperv.pathutils
-import nova.virt.hyperv.vif
-import nova.virt.hyperv.vmops
-import nova.virt.hyperv.volumeops
 import nova.virt.imagecache
 import nova.virt.libvirt.driver
 import nova.virt.libvirt.imagebackend
@@ -38,6 +33,7 @@ import nova.virt.libvirt.volume.remotefs
 import nova.virt.libvirt.volume.scality
 import nova.virt.libvirt.volume.smbfs
 import nova.virt.libvirt.volume.volume
+import nova.virt.solariszones.driver
 import nova.virt.vmwareapi.driver
 import nova.virt.vmwareapi.images
 import nova.virt.vmwareapi.vif
@@ -63,14 +59,6 @@ def list_opts():
              nova.virt.imagecache.imagecache_opts,
          )),
         ('guestfs', nova.virt.disk.vfs.guestfs.guestfs_opts),
-        ('hyperv',
-         itertools.chain(
-             nova.virt.hyperv.pathutils.hyperv_opts,
-             nova.virt.hyperv.vif.hyperv_opts,
-             nova.virt.hyperv.vmops.hyperv_opts,
-             nova.virt.hyperv.volumeops.hyper_volumeops_opts,
-             nova.virt.hyperv.eventhandler.hyperv_opts
-         )),
         ('libvirt',
          itertools.chain(
              nova.virt.libvirt.driver.libvirt_opts,
@@ -91,6 +79,7 @@ def list_opts():
              nova.virt.libvirt.volume.scality.volume_opts,
              nova.virt.libvirt.volume.smbfs.volume_opts,
          )),
+        ('solariszones', nova.virt.solariszones.driver.solariszones_opts),
         ('vmware',
          itertools.chain(
              [nova.virt.vmwareapi.vim_util.vmware_opts],
